libraries/node-core-library/src/Import.ts

```diff
@@ -83,4 +173,109 @@ export class Import {
     const importLazyLocal: (moduleName: string) => unknown = importLazy(require);
     return importLazyLocal(moduleName);
   }
+
+  /**
+   * Resolve a module, from the perspective of a specified folder. This is used to resolve a path that
+   * would be used in a `require(...)` or `import '...'` statement.
+   */
+  public static resolveModule(options: IImportResolveModuleOptions): string {
+    const { modulePath } = options;
+
+    if (path.isAbsolute(modulePath)) {
+      return modulePath;
+    }
+
+    const normalizedRootPath: string = FileSystem.getRealPath(options.baseFolderPath);
+
+    if (modulePath.startsWith('.')) {
+      // This looks like a conventional relative path
+      return path.resolve(normalizedRootPath, modulePath);
+    }
+
+    if (options.includeSystemModules === true && Import._builtInModules.has(modulePath)) {
+      return modulePath;
+    }
+
+    if (options.allowSelfReference === true) {
+      const ownPackage: IPackageDescriptor | undefined = Import._getPackageName(options.baseFolderPath);
+      if (ownPackage && modulePath.startsWith(ownPackage.packageName)) {
+        const packagePath: string = modulePath.substr(ownPackage.packageName.length + 1);
+        return path.resolve(ownPackage.packageRootPath, packagePath);
+      }
+    }
+
+    try {
+      return Resolve.sync(
+        // Append a slash to the package name to ensure `resolve.sync` doesn't attempt to return a system package
+        options.includeSystemModules !== true && modulePath.indexOf('/') === -1
+          ? `${modulePath}/`
+          : modulePath,
+        {
+          basedir: normalizedRootPath,
+          preserveSymlinks: false
+        }
+      );
+    } catch (e) {
+      throw new Error(`Cannot find module "${modulePath}" from "${options.baseFolderPath}".`);
+    }
+  }
+
+  /**
+   * Resolve the path to a package, from the perspective of a specified folder.
```

Comment:
```suggestion
   * Performs module resolution to determine the folder where a package is installed.
   *
   * @remarks
   * Suppose `example` is an NPM package whose entry point is `lib/index.js`:
   * ```ts
   * // Returns "/path/to/project/node_modules/example"
   * Import.resolvePackage({ packageName: 'example' });
   * ```
   *
   * If you need to resolve a module path, use {@link Import.resolveModule} instead:
   * ```ts
   * // Returns "/path/to/project/node_modules/example/lib/index.js"
   * Import.resolveModule({ modulePath: 'example' });
   * ```
   *
   * @returns the absolute path of the package folder.
   * If {@link IImportResolveOptions.includeSystemModules} is specified
   * and a system module is found, then its name is returned without any file path.
```